Source/DafnyCore/AST/Modules/ModuleDefinition.cs

```diff
@@ -214,11 +229,15 @@ public class ModuleDefinition : RangeNode, IAttributeBearingDeclaration, IClonea
       if (IsBuiltinName) {
         compileName = Name;
       } else if (EnclosingModule is { TryToAvoidName: false }) {
-        // Include all names in the module tree path, to disambiguate when compiling
-        // a flat list of modules.
-        // Use an "underscore-escaped" character as a module name separator, since
-        // underscores are already used as escape characters in SanitizeName()
-        compileName = EnclosingModule.GetCompileName(options) + options.Backend.ModuleSeparator + NonglobalVariable.SanitizeName(Name);
+        if (options.Get(LegacyModuleNames)) {
```

Comment:
We should make it so that read translation record options are automatically stored in the module specific options that are attached to ModuleDecl, and then let the `options` here be those module specific options, so that if a Dafny 4.7 project that does not use `--legacy-module-names`, imports a Dafny 4.7 project that does use `--legacy-module-names`, it still works.

Reply:
I had that in mind as a future improvement when I added translation records too.